app/models/period.rb

```diff
@@ -87,6 +97,12 @@ def previous
     end
   end
 
+  def downto(number)
```

Comment:
Ha, took me a bit to understand this, but I get it. I also can't think of a simpler way that isn't imperative,

```ruby
# worse
def downto(number)
  periods = [self]
    
  number.times do
    periods << periods.last.previous
  end

  periods
end
```

Reply:
`downto` is standard enough that I think this is fine 😅

Reply:
Yeah, its a ruby stdlib method on a few core classes: https://apidock.com/ruby/Integer/downto